crates/rome_js_analyze/src/semantic_analyzers/correctness/no_unused_variables.rs

```diff
@@ -106,83 +105,169 @@ declare_rule! {
     }
 }
 
+/// Suggestion if the bindnig is unused
+#[derive(Copy, Clone)]
+pub enum SuggestedFix {
+    /// No suggestion will be given
+    NoSuggestion,
+    /// Suggest to prefix the name of the binding with underscore
+    PrefixUnderscore,
+}
+
+fn is_ok_to_be_unused(parent_function: Option<JsAnyParameterParentFunction>) -> bool {
+    matches!(
+        parent_function,
+        Some(
+            // bindings in signatures are ok to not be  used
+            JsAnyParameterParentFunction::TsMethodSignatureClassMember(_)
+            | JsAnyParameterParentFunction::TsCallSignatureTypeMember(_)
+            | JsAnyParameterParentFunction::TsConstructSignatureTypeMember(_)
+            | JsAnyParameterParentFunction::TsConstructorSignatureClassMember(_)
+            | JsAnyParameterParentFunction::TsMethodSignatureTypeMember(_)
+            | JsAnyParameterParentFunction::TsSetterSignatureClassMember(_)
+            | JsAnyParameterParentFunction::TsSetterSignatureTypeMember(_)
+            // bindings in function types are ok to not be used
+            | JsAnyParameterParentFunction::TsFunctionType(_)
+            // binding in declare are ok to not be used
+            | JsAnyParameterParentFunction::TsDeclareFunctionDeclaration(_)
+        )
+    )
+}
+
+fn is_public_or_private(parameter: TsPropertyParameter) -> Option<bool> {
+    for modifier in parameter.modifiers().into_iter() {
```

Comment:
Nit: `AstNodeList.iter()` and `AstNodeList::into_iter` return the same iterator
```suggestion
    for modifier in parameter.modifiers() {
```

Reply:
Done.